integration/client/container_test.go

```diff
@@ -364,6 +364,67 @@ func withStdout(stdout io.Writer) cio.Opt {
 	}
 }
 
+func TestContainerWait(t *testing.T) {
+	t.Parallel()
+
+	client, err := newClient(t, address)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer client.Close()
+
+	var (
+		image       Image
+		ctx, cancel = testContext(t)
+		id          = t.Name()
+	)
+	defer cancel()
+
+	image, err = client.GetImage(ctx, testImage)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	container, err := client.NewContainer(ctx, id, WithNewSnapshot(id, image), WithNewSpec(oci.WithImageConfig(image), longCommand))
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer container.Delete(ctx)
+
+	task, err := container.NewTask(ctx, empty())
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer task.Delete(ctx)
+
+	ctx2, cancle2 := context.WithCancel(ctx)
+	cancle2()
+	statusErrC, _ := task.Wait(ctx2)
```

Comment:
Next action (in the followup): ensure task.Wait has been invoked and then cancel context.